### ISSUE TITLE
Feature request: add support to remove custom parameters from url

### ISSUE THREAD

**Issue by Enkidu70** [leon-cleaning-services/leon#141]
_In followup to #44 & #70:_

## Minimum feature request:
Please implement a solution to make Leon able remove a specific set of parameters from URLs, like
* clever*

Many websites use same analytic software for campain & user tracking. The name prefix above are a very common one and Léon should be able to generally remove them, with all of their variations.

### Exampel URL:
https://www.ruhr24.de/service/krankenkasse-beitrag-steigt-aok-barmer-tk-guenstig-24h-r24-dak-versicherung-deutschland-zusatz-91998261.html#?cleverPushBounceUrl=https%3A%2F%2Fwww.ruhr24.de%2Fservice%2F&cleverPushNotificationId=yMlhjgj4gsLdgfjhjhgfYdnGCYDEyc
### Real necessary:
https://www.ruhr24.de/service/krankenkasse-beitrag-steigt-aok-barmer-tk-guenstig-24h-r24-dak-versicherung-deutschland-zusatz-91998261.html#

## Maximum feature request:
In addition to min. feature request, it would be very helpful to be able to add own parameter patterns.

E.g. like "*_ref=*"

## Real high end idea:
Enable community of active Léon users to provide their custom parameter list.

## Why?
Keep URL short, keep privacy and empower the user to remove unwanted stuff.

Tnx in advance,
Enkidu

**Comment by svenjacobs** [comment]
Hello @Enkidu70, thanks for your feedback.

Regarding `utm*` parameters, these should already be removed by the Google Analytics sanitizer. Is this not the case?

**Comment by Enkidu70** [comment]
Sorry, concerning utm* you are right. Don't know what went wrong on my side...

**Comment by svenjacobs** [comment]
I agree that custom parameters would be a nice feature. However this would also be the most complex feature to date. I think we should start small and first only allow configuring **exact** parameter names which should be removed. Regular expressions add quite some complexity and possible problems (see my concerns [here](https://github.com/svenjacobs/leon/issues/70)). Maybe later we can even add an option to import/export custom parameters?

**Comment by Enkidu70** [comment]
Well... _Exact_ seems to be a lot of names concerning "clever" parameters. :rofl:

What about adding fixed pattern like "clever*=" as new filter? :pray: :grinning:

**Comment by svenjacobs** [comment]
@Enkidu70 I've been [thinking](https://github.com/svenjacobs/leon/discussions/158) about the custom rules feature. Please provide your feedback in the discussion item.

**Comment by svenjacobs** [comment]
This has been implemented in #797. While it's not strictly custom rules, the new UI allows to select or deselect parts of an URL to be removed.
